aws-logs-metricfilter/src/main/java/software/amazon/logs/metricfilter/DeleteHandler.java

```diff
@@ -0,0 +1,64 @@
+package software.amazon.logs.metricfilter;
+
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DeleteMetricFilterRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.DeleteMetricFilterResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.InvalidParameterException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.OperationAbortedException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.ServiceUnavailableException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnResourceConflictException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class DeleteHandler extends BaseHandlerStd {
+    private Logger logger;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final CallbackContext callbackContext,
+        final ProxyClient<CloudWatchLogsClient> proxyClient,
+        final Logger logger) {
+
+        this.logger = logger;
+
+        final ResourceModel model = request.getDesiredResourceState();
+
+        this.logger.log(String.format("Trying to delete model %s", model.getPrimaryIdentifier()));
+
+        return ProgressEvent.progress(model, callbackContext)
+            .then(progress ->
+                proxy.initiate("AWS-Logs-MetricFilter::Delete", proxyClient, model, callbackContext)
+                    .translateToServiceRequest(Translator::translateToDeleteRequest)
+                    .makeServiceCall(this::deleteResource)
+                    .success());
```

Comment:
no need in `ProgressEvent.progress()` since u use only one callchain.
```java
return proxy.initiate("AWS-Logs-MetricFilter::Delete", proxyClient, model, callbackContext)
                    .translateToServiceRequest(Translator::translateToDeleteRequest)
                    .makeServiceCall(this::deleteResource)
                    .success()
```